src/user/dto/create-user.dto.ts

```diff
@@ -0,0 +1,7 @@
+import { Role } from "src/shared/enums/user-role.enum";
+
+export class CreateUserDto {
```

Comment:
You are using this `class` as a `type` or TS `interface`. I don't know nest so well but doesn't it allow accepting those two instead of `class` in its place? Besides the `Dto` in the name sounds a bit `redundant`. The place and situation you use it should be clear enough to call it just `CreateUser` or `CreateUserModel` IMO, but I may be wrong

Reply:
It is the way the code is written in Nest’s documentation. Classes can utilize validators as seen here: https://docs.nestjs.com/techniques/validation#auto-validation

The ‘Dto’ suffix is also used widely in the documentation. Example: https://docs.nestjs.com/controllers#request-payloads